nextflow/EVAImport/main.nf

```diff
@@ -0,0 +1,301 @@
+      """
+  
+  else if(species == "rattus_norvegicus")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "rat" --species ${species} --registry ${registry} --host ${host} --port ${port} --user 'ensro' --db_name $dbname
+      """
+  else 
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+      """
+}
+
+process run_variation_set {
+  input:
+  val wait
+  val var_set_script
+  val files_path
+  val filenames
+  val set_names
+  val species
+  val registry
+
+  output: val 'ok'
+
+  exec:
+
+  // variable my_set has to be before any if statement
+  // related to this issue: https://github.com/nextflow-io/nextflow/issues/804
+  def my_set = set_names.get(species)
+  for (String name : my_set) {
+    def input_file = filenames.get(name)
+    def command = "perl ${var_set_script} -load_file ${files_path}${input_file} -registry ${registry} -species ${species} -variation_set ${name}"
+    command.execute()
+  }
+}
+
+// Post-process variation feature sets
+// Populates variation_feature.variation_set_id
+process run_variation_set_2 {
+  input:
+  val wait
+  val var_set_script_2
+  val species
+  val registry
+
+  output: val 'ok'
+
+  script:
+
+  """
+  perl ${var_set_script_2} -registry_file ${registry} -species ${species}
+  """
+}
+
+process prepare_tables {
+  input:
+  val copy_script
+  val host
+  val dbname
+  val old_dbname
+
+  output: val 'ok'
+
+  script:
+  def new_host = "${host}-ensadmin"
+
+  """
+  sh ${copy_script} -h ${new_host} -o ${old_dbname} -n ${dbname}
+  """
+}
+
+process run_citations {
+  input:
+  val wait
+  path citations_script
+  val species
+  val registry
+  val file
+
+  output: val 'ok'
+
+  script:
+
+  """
+  perl ${citations_script} -load_file ${file} -registry ${registry} -species ${species}
+  """
+}
+
+
+workflow {
+  prepare_tables(copy_tables_script, params.host, params.dbname, params.old_dbname)
+
+  run_eva(prepare_tables.out, file(eva_script), command_to_run, params.merge_all_types, params.fork, params.sort_vf, params.chr_synonyms, params.remove_prefix, params.output_file)
+
+  run_variant_synonyms(run_eva.out, file(var_syn_script), params.source, params.species, params.var_syn_file, params.registry, params.old_host, params.old_port, params.old_dbname)
+
+  if(set_names[params.species]) {
+    run_variation_set(run_variant_synonyms.out, var_set_script, files_path, filenames, set_names, params.species, params.registry)
+    run_variation_set_2(run_variation_set.out, var_set_script_2, params.species, params.registry)
+  }
+
+  if(params.citations_file) {
+    run_citations(run_variant_synonyms.out, file(citations_script), params.species, params.registry, params.citations_file)
+  }
+}
```

Comment:
Can we have a `workflow.onComplete` code block upon completion of the workflow similar to this [example](https://github.com/Ensembl/ensembl-variation/blob/18a12b68c7fd8302744b0e99b105c8d2e6f05e92/nextflow/ProteinFunction/main.nf#LL186C1-L186C20)?